### PR TITLE
Rework Bash import

### LINKED ISSUE #745: Bash history import issues
I spotted two issues while trying to import my Bash history:

1. Import ordering is incorrectly reversed.
2. `.bash_history` can be configured to store timestamps, a case not handled.

---

## Ordering

In the following snippet, lines from the top of the file will receive the newest timestamp:

https://github.com/cyqsimon/atuin/blob/d46e3ad47d03e53232269973806fb9da2248ba19/atuin-client/src/import/bash.rs#L42-L67

This is incorrect for `.bash_history` - new items are appended to the end of the file, not the start. So the first lines are in fact the oldest commands.

## Stored timestamps

The current import code for Bash assumes that each line is a command. However if `HISTTIMEFORMAT` is set, then a timestamp is recorded for each command (see `man bash`). So `.bash_history` would look something like this:

```
#1672918999
git reset
#1672919006
git clean -dxf
#1672919020
cd ../
```

This case is currently not handled, so the timestamp lines are getting falsely-imported as commands.

---

I am working on a patch that will fix both these issues, so give me a little while and I'll submit a PR.

The reason I'm posing this issue mostly concerns the extent of the ordering problem. Did it happen because
- the original author used a different shell that does store the newest commands at the start of its history file, and wrongly assumed the same for Bash?
- or the original author made a simple arithmetic mistake, and this is an issue for the import code of all shells?

In other words, I'm not that familiar with other shells, so does the import code for them need fixing as well?

### PR DESCRIPTION
**Pull request by cyqsimon**
Closes #745
- Imported history is now ordered correctly
- Timestamps (when `HISTTIMEFORMAT` is set) are handled correctly

### CODE REVIEW

**Comment by cyqsimon** [comment]
Yeah... `utils::tests::test_config_dir` this test is failing spuriously on my local machine as well. It's not related to this PR though so I'll issue a rerun.

File: atuin-client/src/import/bash.rs [hunk 1]
```diff
@@ -32,64 +34,110 @@ impl Importer for Bash {
...
-        for (i, b) in unix_byte_lines(&self.bytes).enumerate() {
-            let s = match std::str::from_utf8(b) {
-                Ok(s) => s,
-                Err(_) => continue, // we can skip past things like invalid utf8
-            };
-
-            if let Some(s) = s.strip_suffix('\\') {
-                line.push_str(s);
-                line.push_str("\\\n");
-            } else {
```
[the remark this answers is not in the document]
Comment (ellie) [reply atuin-client/src/import/bash.rs:51]: I see - makes sense!

**Review by ellie — APPROVED** [review]
Thank you! Code looks good to me, though I can't test it super easily given my lack of bash history. Will see if I can get some going before the next release

File: atuin-client/src/import/bash.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (llimllib) [reply atuin-client/src/import/bash.rs:51]: oh maybe that's just because it's inside a quote though!

[the remark this answers is not in the document]
Comment (cyqsimon) [reply atuin-client/src/import/bash.rs:51]: > Bash definitely records history with line continuation characters for me:

Yeah I realised that yesterday. Another example is when you use here document syntax. See #167 and my response.

TLDR, the way it's done right now is certainly far from ideal, but it's "good enough"™️ in my book for now. The way Bash stores history is a *huge* mess anyways, so I'd much rather spend a little time and get 90% than burn my brain out and get 99%.

If any user ***really** cares about their history being 100% clean (why?), they can always go clean up the DB file manually.

[the remark this answers is not in the document]
Comment (llimllib) [reply atuin-client/src/import/bash.rs:51]: works for me, and it's what I did when I imported.

Thanks again for the work!

**Pull request merged**